### PR TITLE
dvdplayer: improve audio sync

### PR DESCRIPTION
**Pull request by FernetMenta**
fixes audio dropouts in particular for PVR

### CODE REVIEW

File: xbmc/cores/dvdplayer/DVDPlayerAudio.cpp [hunk 1]
```diff
@@ -659,71 +659,77 @@ void CDVDPlayerAudio::HandleSyncError(double duration)
...
 
-        proportional = m_error / DVD_TIME_BASE / proportionaldiv;
-      }
-      m_resampleratio = 1.0 / m_pClock->GetClockSpeed() + proportional + m_integral;
+    //on big errors use more proportional
+    if (fabs(m_error / DVD_TIME_BASE) > 0.0)
+    {
+      double proportionaldiv = PROPORTIONAL * (PROPREF / fabs(m_error / DVD_TIME_BASE));
+      if (proportionaldiv < PROPDIVMIN) proportionaldiv = PROPDIVMIN;
+      else if (proportionaldiv > PROPDIVMAX) proportionaldiv = PROPDIVMAX;
```
Comment (AlwinEsch) [inline xbmc/cores/dvdplayer/DVDPlayerAudio.cpp]: Can you change it then here to the code style.
```C++
    double proportionaldiv = PROPORTIONAL * (PROPREF / fabs(m_error / DVD_TIME_BASE));
    if (proportionaldiv < PROPDIVMIN) 
      proportionaldiv = PROPDIVMIN;
    else if (proportionaldiv > PROPDIVMAX)
      proportionaldiv = PROPDIVMAX;
```

Comment (FernetMenta) [inline xbmc/cores/dvdplayer/DVDPlayerAudio.cpp]: good idea, will do

**Comment by FernetMenta** [comment]
jenkins build this please

**Comment by FernetMenta** [comment]
I added  e80515d which fixes an endless loop of audio stall / buffering / stall which can happen after a bad signal for PVR. audio can't handle large sync errors by itself. player needs to re-snyc

**Comment by FernetMenta** [comment]
jenkins build this please

**Comment by FernetMenta** [comment]
jenkins build this please

**Comment by FernetMenta** [comment]
jenkins build this please

**Comment by FernetMenta** [comment]
there is still some issues left for FLAC which has packets of 96ms.

**Pull request merged**